internal/core/service/user_service_test.go

```diff
@@ -0,0 +1,228 @@
+		mockRepo := &MockUserRepository{Users: Users}
+		service := NewUserService(mockRepo)
+
+		result, err := service.Find(nil, validUser.ID)
+		if err != nil {
+			t.Errorf("error finding user: %v", err)
+		}
+		if !reflect.DeepEqual(result, &validUser) {
+			t.Errorf("expected: %v, result: %v", validUser, result)
+		}
+	})
+	t.Run("Should return error with user not found", func(t *testing.T) {
+		mockRepo := &MockUserRepository{}
+		service := NewUserService(mockRepo)
+
+		result, err := service.Find(nil, validUser.ID)
+		if result != nil {
+			t.Errorf("should return userNotFound")
+		}
+		if err == nil {
+			t.Errorf("expected: %v, result: %v", ErrUserNotFound, err)
+		}
+	})
+}
+func TestUpdateUser(t *testing.T) {
+	t.Run("Update valid user", func(t *testing.T) {
+		Users := []model.User{validUser}
+		mockRepo := &MockUserRepository{Users: Users}
+		service := &UserService{repo: mockRepo}
+
+		updatedUser := model.User{
+			ID:        validUser.ID,
+			FirstName: "Doe",
+			LastName:  "NewUserService",
+			Email:     "new@doe.com",
+			Age:       23,
+		}
+
+		result, err := service.Update(nil, updatedUser)
+		if err != nil {
+			t.Errorf("error finding user: %v", err)
+		}
+		if !reflect.DeepEqual(result, &updatedUser) {
+			t.Errorf("expected: %v, result: %v", updatedUser, result)
+		}
+		if len(mockRepo.Users) != 1 {
+			t.Errorf("user not updated properly in mock repository")
+		}
+	})
+	t.Run("return error on update with first and lastname that already exists", func(t *testing.T) {
+		updatedUser := model.User{
+			ID:        primitive.NewObjectID().Hex(),
+			FirstName: "John",
+			LastName:  "Doe Second",
+			Email:     "new@doe.com",
+			Age:       25,
+		}
+		Users := []model.User{validUser, updatedUser}
+		mockRepo := &MockUserRepository{Users: Users}
+		service := &UserService{repo: mockRepo}
+
+		changedValidUser := model.User{
+			ID:        validUser.ID,
+			FirstName: updatedUser.FirstName, // already exists
+			LastName:  updatedUser.LastName,  // already exists
+			Email:     "new@doe.com",
+			Age:       25,
+		}
+
+		_, err := service.Update(nil, changedValidUser)
+		if err == nil {
+			t.Errorf("should return error updating user: %v", err)
+		}
+		if len(mockRepo.Users) != 2 {
+			log.Println(mockRepo.Users)
+			t.Errorf("duplicate user should not be updated in mock repository")
+		}
+	})
+	t.Run("return error with existing user first and last name", func(t *testing.T) {
+		Users := []model.User{validUser}
+		mockRepo := &MockUserRepository{Users: Users}
+		service := &UserService{repo: mockRepo}
+
+		updatedUser := model.User{
+			ID:        primitive.NewObjectID().Hex(),
+			FirstName: "John",
+			LastName:  "Doe",
+			Email:     "new@doe.com",
+			Age:       23,
+		}
+
+		result, err := service.Update(nil, updatedUser)
+		if err == nil || result != nil {
+			t.Errorf("update should fail: %v", err)
+		}
+	})
+}
+func TestUserValidation(t *testing.T) {
+	t.Run("Validate user", func(t *testing.T) {
```

Comment:
Name the test with a more descriptive sentence. Like "Validate user without error"